Source/player.cpp

```diff
@@ -3156,7 +3147,8 @@ StartPlayerKill(int pnum, int earflag)
 			drawhpflag = true;
 
 			if (pcurs >= CURSOR_FIRSTITEM) {
-				DeadItem(player, &player.HoldItem, { 0, 0 });
+				DeadItem(player, player.HoldItem, { 0, 0 });
+				player.HoldItem._itype = ItemType::None;
 				NewCursor(CURSOR_HAND);

```

Comment:
This actually highlights an issue, if a player is holding an item in the cursor on death the item will get deleted :o

We should probably try and drop the item first, but if you prefer we can wait with that till a later PR